### PR TITLE
New shared IP article needing stakeholder review

### PR DESCRIPTION
**Pull request by catlook**

### CODE REVIEW

**Comment by catlook** [comment]
@stephamon take a look please

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 1]
```diff
@@ -0,0 +1,621 @@
...
+
+This article describes how to run an Apache2 web server on a pair of
+cloud servers with a shared public IPv4 IP address. The shared IP
+address ensures that users of services don't need to change addresses
+when the second server seamlessly takes over from the first after the
+first server goes down.
+
+The use case in this article shows an Apache web server deployed on two
+Ubuntu cloud servers with a shared IP address on PublicNet.  The shared 
+IP address is used to reach the web servers. The use case uses a
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: "The use case uses" is kind of repetitive. Can we rephrase this sentence somehow? 

Maybe "The protocol is a heartbeat that determines which cloud server..."?

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
+    3. Configure apache2 and set up the heartbeat on the servers
+
+    4. Test the configuration
+
+### Before you begin
+
+You should have an authentication token and the ability to use cURL in
+your environment.
+
+### Step 1: Create two servers in the same publicIPZoneId
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: ### Create two servers in the same publicIPZoneId

(minus "Step 1:")

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 3]
```diff
@@ -0,0 +1,621 @@
...
+
+#### Perform the following steps from your local computer.
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: No header here.

"Perform the following steps from your local computer:"

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 4]
```diff
@@ -0,0 +1,621 @@
...
+
+#### 1. Boot the master server. Save the server ID for future reference.
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: These steps shouldn't be headers. Just a simple numbered list will do.

So:

```
Perform the following steps from your local computer:

1. Boot the master server...
2. Get the details...
3. Boot the slave server...
```

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 5]
```diff
@@ -0,0 +1,621 @@
...
+
+*Request:*
+
+```curl -s https://dfw.servers.api.rackspacecloud.com/v2/5831008/servers <br>
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: If someone copy/pasted this code, would the <br> tag break it? Don't you use a `\` to break up long lines of code?

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 6]
```diff
@@ -0,0 +1,621 @@
...
+      "id": "96bbd712-0f64-4146-bfb2-b2bd91f20319",
+      "links": [
+         {"href": "https://dfw.servers.api.rackspacecloud.com/v2/5831008/servers/96bbd712-0f64-4146-bfb2-b2bd91f20319", "rel": "self"},
+         {"href": "https://dfw.servers.api.rackspacecloud.com/5831008/servers/96bbd712-0f64-4146-bfb2-b2bd91f20319", "rel": "bookmark"}
+      ],
+      "adminPass": "P6dW6YLk4pLQ"
+   }
+}
+```
+#### 2. Get the details for the master server. Save the publicIPZoneId for future reference.
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: Same thing here as above about the header and the "Save the publicIPZoneId" sentence.

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 7]
```diff
@@ -0,0 +1,621 @@
...
+      "name": "isol1",
+      "progress": 100,
+      "status": "ACTIVE",
+      "tenant_id": "5831008",
+      "updated": "2015-07-27T20:03:46Z",
+      "user_id": "207638"
+   }
+}
+```
+#### 3. Boot the slave server in same publicIPZoneId as the master server by using a scheduler hint. Save the server ID
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: Same thing here as above about the header and the "Save the server ID" sentence.

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 8]
```diff
@@ -0,0 +1,621 @@
...
+      "id": "ffec9d55-2d54-4718-bc3a-0d47fb8c52c1",
+      "links": [
+         {"href": "https://dfw.servers.api.rackspacecloud.com/v2/5831008/servers/ffec9d55-2d54-4718-bc3a-0d47fb8c52c1", "rel": "self"},
+         {"href": "https://dfw.servers.api.rackspacecloud.com/5831008/servers/ffec9d55-2d54-4718-bc3a-0d47fb8c52c1", "rel": "bookmark"}
+      ],
+      "adminPass": "q5dmoSkbYLF6"
+   }
+}
+```
+#### 4: Get the details for the slave server and confirm that slave server is in the same publicIPZoneId as master server.
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: Same as above about the header.

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 9]
```diff
@@ -0,0 +1,621 @@
...
+      "name": "isol2",
+      "progress": 100,
+      "status": "ACTIVE",
+      "tenant_id": "5831008",
+      "updated": "2015-07-27T20:04:39Z",
+      "user_id": "207638"
+   }
+}
+```
+### Step 2: Create a shared IP address and associate it with both servers
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: `### Create a shared IP address and associate it with both servers`

(minus "Step 2:")

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 10]
```diff
@@ -0,0 +1,621 @@
...
+
+Perform the following steps from your local computer.
+
+#### 1. Get port IDs for both servers. Filter the list ports operation by the PublicNet (network ID that is all zeroes).
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: Same thing here in the previous section. No headers on these steps, just a simple numbered list.

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 11]
```diff
@@ -0,0 +1,621 @@
...
+```
+*Response:*
+
+```{"ip_association": {
+      "id": "e201f500-6d57-4901-b7a0-3842a3a32207",
+      "address": "10.23.233.31"
+   }
+}
+```
+### Step 3: Configure Apache2 and set up the heartbeat on the servers
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: `### Configure Apache2 and set up the heartbeat on the servers`

(minus "Step 3:")

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 12]
```diff
@@ -0,0 +1,621 @@
...
+
+Perform the following steps on the master and slave servers. Each step
+indicates where to perform the step.
+
+#### 1. (Master) Install Apache on the master server by running the following commands:
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: Again, no headers, just a numbered list.

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 13]
```diff
@@ -0,0 +1,621 @@
...
+```
+#### 14. (Master) Restart Apache on the master server by running the following command:
+
+```sudo service apache2 restart
+```
+#### 15. (Slave) Restart Apache on the slave server by running the following command:
+
+```sudo service apache2 restart
+```
+### Step 4: Test the configuration
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: `### Test the configuration`

File: content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md [hunk 14]
```diff
@@ -0,0 +1,621 @@
...
+
+Perform the following steps on the master and slave servers and your
+local computer. Each step indicates where to perform the step.
+
+#### 1. (Master) Validate the eth0 interface configured with the shared IP address on the master server by running
```
Comment (stephamon) [inline content/cloud-networks/set-up-apache2-web-server-on-two-instances-with-shared-ip.md]: No headers, just a numbered list :)

**Comment by stephamon** [comment]
@catlook I've got some formatting changes mostly.

tl;dr: The big steps (1-4) can lose the "Step #" part, and the numbered lists shouldn't have the h4 level headers. 

Probably, Kelly should take a pass when you're ready. :)

**Comment by catlook** [comment]
close this.  when i applied my fixes, i accidentally put n new PR.
